Composer/packages/adaptive-form/src/components/fields/TextField/TextField.base.tsx

```diff
@@ -0,0 +1,624 @@
+   *
+   * - If there is no validation error or we have not validated the input value, errorMessage is an empty string.
+   * - If we have done the validation and there is validation error, errorMessage is the validation error message.
+   */
+  private get errorComponentMessage(): string | JSX.Element {
+    const { errorMessage = this.state.errorMessage } = this.props;
+    return errorMessage || '';
+  }
+
+  /**
+   * If a custom description render function is supplied then treat description as always available.
+   * Otherwise defer to the presence of description or error message text.
+   */
+  private get isDescriptionAvailable(): boolean {
+    const props = this.props;
+    return !!(props.onRenderDescription || props.description || this.errorComponentMessage);
+  }
+
+  private get componentClassNames(): IProcessedStyleSet<ITextFieldStyles> {
+    const {
+      borderless,
+      className,
+      disabled,
+      iconProps,
+      inputClassName,
+      label,
+      multiline,
+      required,
+      underlined,
+      resizable,
+      theme,
+      styles,
+      autoAdjustHeight,
+    } = this.props;
+    const { isFocused } = this.state;
+    const errorMessage = this.errorComponentMessage;
+
+    return getClassNames(styles!, {
+      theme: theme!,
+      className,
+      disabled,
+      focused: isFocused,
+      required,
+      multiline,
+      hasLabel: !!label,
+      hasErrorMessage: !!errorMessage,
+      borderless,
+      resizable,
+      hasIcon: !!iconProps,
+      underlined,
+      inputClassName,
+      autoAdjustHeight,
+    });
+  }
+
+  private _renderTextArea = (): React.ReactElement<React.HTMLAttributes<HTMLAreaElement>> => {
+    const textAreaProps = getNativeProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>>(
+      this.props,
+      textAreaProperties
+    );
+
+    const ariaLabelledBy = this.props['aria-labelledby'] || (this.props.label ? this._labelId : undefined);
+    /*-----added by composer-----*/
+    // use the default value to replace the state value
+    const valueProps = this.isComponentControlled
+      ? { value: this.props.value }
+      : { defaultValue: this.props.defaultValue };
+    /*-----added by composer-----*/
+    return (
+      <textarea
+        id={this.componentId}
+        {...textAreaProps}
+        {...valueProps}
+        ref={this._textElement as React.RefObject<HTMLTextAreaElement>}
+        aria-describedby={this.isDescriptionAvailable ? this._descriptionId : this.props['aria-describedby']}
+        aria-invalid={!!this.errorComponentMessage}
+        aria-label={this.props.ariaLabel}
+        aria-labelledby={ariaLabelledBy}
+        className={this.componentClassNames.field}
+        defaultValue={this.props.defaultValue ?? ''}
+        readOnly={this.props.readOnly}
+        onBlur={this._onBlur}
+        onChange={this._onInputChange}
+        onFocus={this._onFocus}
+        onInput={this._onInputChange}
+      />
+    );
+  };
```

Comment:
When the component is uncontrolled I believe you have _**defaultValue**_ 3 times: 

- Line 464 from getNativeProps
- Line 472 from valueProps 
- Line 487 from props directly

When the component is controlled you have it two times:

- Line 464 from getNativeProps
- Line 487 from props directly

When the component is uncontrolled I believe you have _**value**_ 1 time: 

- Line 464 from getNativeProps

When the component is controlled you have it two times:

- Line 464 from getNativeProps
- Line 472 from valueProps 


Could you explain why you removed the third argument of getNativeProps from the Fabric code? This 3rd argument specifies what properties to exclude:

`(props: Record<string, any>, allowedPropNames: string[] | Record<string, number>, excludedPropNames?: string[])`

Therefore it looks like both _**value**_ and _**defaultValue**_ are always passed along and valueProps does nothing.

Have you explored something like this instead?

![image](https://user-images.githubusercontent.com/66701106/94056624-671a0600-fd93-11ea-87fc-4c3045561aa6.png)

Reply:
Thanks @LouisEugeneMSFT .I think you are right and I updated this part.